src/pages/Lending/Pool/components/Borrow/BorrowConfirm.tsx

```diff
@@ -99,16 +104,63 @@ export const BorrowConfirm = ({ collateralAssetId, depositAmount }: BorrowConfir
   // so we have a safety to not refetch quotes while borrow is pending
   // perhaps a shared react-query mutation hook would make sense in handleSend(), so we have a way to introspect pending status
   // from input components and disable inputs as well?
-  const { data: lendingQuoteData, isLoading: isLendingQuoteLoading } = useLendingQuoteQuery({
+  const {
+    data,
+    isLoading: isLendingQuoteLoading,
+    isError: isLendingQuoteError,
+  } = useLendingQuoteQuery({
     collateralAssetId,
     borrowAssetId,
     depositAmountCryptoPrecision: depositAmount ?? '0',
   })
 
+  const lendingQuoteData = isLendingQuoteError ? null : data
+
   const chainAdapter = getChainAdapterManager().get(fromAssetId(collateralAssetId).chainId)
 
   const selectedCurrency = useAppSelector(selectSelectedCurrency)
 
+  const collateralAccountFilter = useMemo(
+    () => ({ accountId: collateralAccountId }),
+    [collateralAccountId],
+  )
+  const collateralAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, collateralAccountFilter),
+  )
+  const collateralAccountType = collateralAccountMetadata?.accountType
+  const collateralBip44Params = collateralAccountMetadata?.bip44Params
+
+  const getFromAddress = useCallback(async () => {
+    if (!(wallet && chainAdapter && collateralBip44Params)) return null
+
+    return isUtxoChainId(fromAccountId(collateralAccountId).chainId)
+      ? await getThorchainLendingPosition({
+          accountId: collateralAccountId,
+          assetId: collateralAssetId,
+        })
+          .then(position => {
+            if (!position) throw new Error(`No position found for assetId: ${collateralAssetId}`)
+          })
+          .catch(async () => {
+            const firstReceiveAddress = await chainAdapter.getAddress({
+              wallet,
+              accountNumber: collateralBip44Params.accountNumber,
+              accountType: collateralAccountType,
+              index: 0,
+            })
+
+            return firstReceiveAddress
+          })
```

Comment:
do we want to stick with the async IIFE approach or chained then/catch here?

Reply:
Will most likely keep it that way for now to keep it consistent with the savers implementation!
Added a TODO for the unification across savers/lending bf5d080287